src/test/java/seedu/toluist/commons/util/StringUtilTest.java

```diff
@@ -148,5 +150,54 @@ public void getDetails_nullGiven_assertionError() {
         StringUtil.getDetails(null);
     }
 
+    //---------------- Tests for splitIndexes --------------------------------------
 
+    @Test
+    public void splitIndexes_obtainIndexesToEndIndex() {
+        List<Integer> actual = StringUtil.splitIndexes(" - 3", 8);
+        List<Integer> expected = Arrays.asList(1, 2, 3);
+        assertTrue(actual.equals(expected));
+    }
+
+    @Test
+    public void splitIndexes_obtainIndexesFromStartIndex() {
+        List<Integer> actual = StringUtil.splitIndexes(" 3 -", 8);
+        List<Integer> expected = Arrays.asList(3, 4, 5, 6, 7, 8);
+        assertTrue(actual.equals(expected));
+    }
+
+    @Test
+    public void splitIndexes_obtainIndexesFromStartIndexToEndIndex() {
+        List<Integer> actual = StringUtil.splitIndexes("3-5", 8);
+        List<Integer> expected = Arrays.asList(3, 4, 5);
+        assertTrue(actual.equals(expected));
+    }
+
+    @Test
+    public void splitIndexes_obtainIndexesFromPoorlyFormattedStringIndexes() {
+        List<Integer> actual = StringUtil.splitIndexes("1-2  ,    3-   4   5  -6   , 7   -      8", 8);
+        List<Integer> expected = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8);
+        assertTrue(actual.equals(expected));
+    }
+
+    @Test
+    public void splitIndexes_obtainIndexesFromAssortedIndexes() {
+        List<Integer> actual = StringUtil.splitIndexes("1 3,  7", 8);
+        List<Integer> expected = Arrays.asList(1, 3, 7);
+        assertTrue(actual.equals(expected));
+    }
+
+    @Test
+    public void splitIndexes_obtainIndexesFromAssortedUnorderedIndexes() {
+        List<Integer> actual = StringUtil.splitIndexes("3, 2, 8, 5-7", 8);
+        List<Integer> expected = Arrays.asList(3, 2, 8, 5, 6, 7);
+        assertTrue(actual.equals(expected));
+    }
+
+    @Test
+    public void splitIndexes_obtainIndexesComplexStringIndexes() {
+        List<Integer> actual = StringUtil.splitIndexes("- 3, 5, 7 - 12", 8);
+        List<Integer> expected = Arrays.asList(1, 2, 3, 5, 7, 8);
+        assertTrue(actual.equals(expected));
+    }
```

Comment:
in particular do u have any test cases u wanna try?
for now i have 
"banana"
"1 banana 3"
"1 3 banana"
The algorithm terminates the moment it reaches invalid state, so the output is [], [1], [1,3] respectively